src/multivariate.jl

```diff
@@ -75,20 +105,26 @@ function MvNormal(
 end
 
 # dense mean, diagonal covariance
-MvNormal(m::TrackedVector{<:Real}, σ::TrackedVector{<:Real}) = TuringDiagNormal(m, σ)
-MvNormal(m::TrackedVector{<:Real}, σ::AbstractVector{<:Real}) = TuringDiagNormal(m, σ)
-MvNormal(m::TrackedVector{<:Real}, σ::Vector{<:Real}) = TuringDiagNormal(m, σ)
-MvNormal(m::AbstractVector{<:Real}, σ::TrackedVector{<:Real}) = TuringDiagNormal(m, σ)
+MvNormal(m::TrackedVector{<:Real}, σ::TrackedVector{<:Real}) = TuringDiagMvNormal(m, σ)
+MvNormal(m::TrackedVector{<:Real}, σ::AbstractVector{<:Real}) = TuringDiagMvNormal(m, σ)
+MvNormal(m::TrackedVector{<:Real}, σ::Vector{<:Real}) = TuringDiagMvNormal(m, σ)
+MvNormal(m::AbstractVector{<:Real}, σ::TrackedVector{<:Real}) = TuringDiagMvNormal(m, σ)
 
 # dense mean, constant variance
-MvNormal(m::TrackedVector{<:Real}, σ::TrackedReal) = MvNormal(m, fill(σ, length(m)))
-MvNormal(m::TrackedVector{<:Real}, σ::Real) = MvNormal(m, fill(σ, length(m)))
-MvNormal(m::AbstractVector{<:Real}, σ::TrackedReal) = MvNormal(m, fill(σ, length(m)))
+MvNormal(m::TrackedVector{<:Real}, σ::TrackedReal) = TuringScalMvNormal(m, σ)
+MvNormal(m::TrackedVector{<:Real}, σ::Real) = TuringScalMvNormal(m, σ)
+MvNormal(m::AbstractVector{<:Real}, σ::TrackedReal) = TuringScalMvNormal(m, σ)
 
 # dense mean, constant variance
-MvNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:TrackedReal}) = MvNormal(m, A.λ)
-MvNormal(m::AbstractVector{<:Real}, A::UniformScaling{<:TrackedReal}) = MvNormal(m, A.λ)
-MvNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:Real}) = MvNormal(m, A.λ)
+function MvNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:TrackedReal})
+    return MvNormal(m, sqrt(A.λ))
+end
+function MvNormal(m::AbstractVector{<:Real}, A::UniformScaling{<:TrackedReal})
+    return MvNormal(m, sqrt(A.λ))
+end
+function MvNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:Real})
+    return MvNormal(m, sqrt(A.λ))
+end
```

Comment:
I think the three constructors above that create an `MvNormal` instance might not be needed anymore. Hopefully that just works with the updated constructors in Distributions 0.22.

Reply:
I think it's needed to avoid calling `ScalMat` in https://github.com/JuliaStats/Distributions.jl/blob/70f3d97fca49b0abfa154f2f29c7b4f4a90b2ff8/src/multivariate/mvnormal.jl#L208. We want `Tracker`-`MvNormal` calls to have nothing to do with `PDMats` to simplify things.

Reply:
Ah sorry, I missed that of course you're just forwarding to the specialized implementation for TrackedReal but never call the constructors in Distributions.